packages/web-components/src/search/search-overlay.ts

```diff
@@ -0,0 +1,192 @@
+    
+    @property({ attribute: 'displayed-at-location' })
+    displayedAtLocation?: string = undefined;
+
+    @property({ attribute: 'search-bar-placeholder' })
+    searchBarPlaceholder: string | null = null;
+
+    @property({ type: Number, attribute: 'number-of-products' })
+    numberOfProducts: number = 5;
+
+    @property({ type: Number, attribute: 'number-of-search-term-predictions' })
+    numberOfSearchTermPredictions: number = 3;
+
+    @property({ attribute: 'no-results-message' })
+    noResultsMessage: string | null = null;
+
+    @property({ type: Number, attribute: 'debounce-time' })
+    debounceTime: number = 250;
+
+    @state()
+    results: SearchResult[] | null = null;
+
+    @state()
+    term: string = '';
+    
+    @state() 
+    selectedIndex = -1;
+
+    @state()
+    isInFocus: boolean = false;
+
+    @state()
+    hasCompletedSearchRequest: boolean = false;
+
+    private debounceTimer: NodeJS.Timeout | null = null;
+    
+    async connectedCallback() {
+        if (!this.displayedAtLocation) {
+            console.error('No displayedAtLocation defined!');
+        }
+        super.connectedCallback();
+    }
+
+    setSearchTerm(term: string) {
+        this.term = term;
+        this.selectedIndex = -1;
+
+        if (!term) {
+            this.results = null; 
+            return;
+        }
+
+        if (this.debounceTimer) {
+            clearTimeout(this.debounceTimer);
+        }
+
+        this.debounceTimer = setTimeout(() => {
+            this.search(term);
+        }, this.debounceTime);
+    }
+
+    handleKeyDown(event: KeyboardEvent): void {
+        if (!this.results) {
+            return;
+        }
+
+        switch (event.key) {
+        case 'ArrowUp':
+            event.preventDefault();
+            this.selectedIndex = Math.max(this.selectedIndex - 1, 0);
+            break;
+        case 'ArrowDown':
+            event.preventDefault();
+            this.selectedIndex = Math.min(this.selectedIndex + 1, this.results.length - 1);
+            break;
+        case 'Tab':
+            event.preventDefault();
+            this.selectedIndex = Math.min(this.selectedIndex + 1, this.results.length - 1);
+            break;
+        case 'Enter':
+            event.preventDefault();
+            this.handleActionOnResult(this.results[this.selectedIndex]);
+            break;
+        }
+    }
+
+    handleActionOnResult(result: SearchResult) {
+        if (result.searchTermPrediction) {
+            this.setSearchTerm(result.searchTermPrediction.term ?? '');
+        }
+
+        if (result.product && result.product.data && 'Url' in result.product.data) {
+            window.location.href = result.product.data['Url'].value ?? '';
+        }
+    }
+
+    async search(searchTerm: string) {
+        const relewiseUIOptions = getRelewiseUIOptions();
+        const searchOptions = getRelewiseUISearchOptions();
+        const settings = getRelewiseContextSettings(this.displayedAtLocation ?? '');
```

Comment:
```suggestion
        const settings = getRelewiseContextSettings(this.displayedAtLocation ?? 'Relewise Product Search Overlay');
```

Jeg har ombestemt mig her. Jeg tror bare vi skal defaulte til nogle generiske navne i stedet for en tom string. Så er det mere tydeligt i My Relewise at implementeringen er forkert :)

Reply:
Check 👍 
Jeg ved ikke om du kan lide de andre defaults jeg har valgt? Det må du meget gerne lige check op på! 🙂